process/economics/economicsData.go

```diff
@@ -215,148 +116,40 @@ func checkValues(economics *config.EconomicsConfig) error {
-	if err != nil {
-		return nil, fmt.Errorf("%w for epoch %d", process.ErrInvalidMaxGasLimitPerMiniBlock, gasLimitSetting.EnableEpoch)
-	}
-
-	gc.maxGasLimitPerMetaBlock, err = strconv.ParseUint(gasLimitSetting.MaxGasLimitPerMetaBlock, conversionBase, bitConversionSize)
-	if err != nil {
-		return nil, fmt.Errorf("%w for epoch %d", process.ErrInvalidMaxGasLimitPerMetaBlock, gasLimitSetting.EnableEpoch)
-	}
-
-	gc.maxGasLimitPerMetaMiniBlock, err = strconv.ParseUint(gasLimitSetting.MaxGasLimitPerMetaMiniBlock, conversionBase, bitConversionSize)
-	if err != nil {
-		return nil, fmt.Errorf("%w for epoch %d", process.ErrInvalidMaxGasLimitPerMetaMiniBlock, gasLimitSetting.EnableEpoch)
-	}
-
-	gc.maxGasLimitPerTx, err = strconv.ParseUint(gasLimitSetting.MaxGasLimitPerTx, conversionBase, bitConversionSize)
-	if err != nil {
-		return nil, fmt.Errorf("%w for epoch %d", process.ErrInvalidMaxGasLimitPerTx, gasLimitSetting.EnableEpoch)
-	}
-
-	gc.extraGasLimitGuardedTx, err = strconv.ParseUint(gasLimitSetting.ExtraGasLimitGuardedTx, conversionBase, bitConversionSize)
-	if err != nil {
-		return nil, fmt.Errorf("%w for epoch %d", process.ErrInvalidExtraGasLimitGuardedTx, gasLimitSetting.EnableEpoch)
-	}
-
-	if gc.maxGasLimitPerBlock < gc.minGasLimit {
-		return nil, fmt.Errorf("%w: maxGasLimitPerBlock = %d minGasLimit = %d in epoch %d", process.ErrInvalidMaxGasLimitPerBlock, gc.maxGasLimitPerBlock, gc.minGasLimit, gasLimitSetting.EnableEpoch)
-	}
-	if gc.maxGasLimitPerMiniBlock < gc.minGasLimit {
-		return nil, fmt.Errorf("%w: maxGasLimitPerMiniBlock = %d minGasLimit = %d in epoch %d", process.ErrInvalidMaxGasLimitPerMiniBlock, gc.maxGasLimitPerMiniBlock, gc.minGasLimit, gasLimitSetting.EnableEpoch)
-	}
-	if gc.maxGasLimitPerMetaBlock < gc.minGasLimit {
-		return nil, fmt.Errorf("%w: maxGasLimitPerMetaBlock = %d minGasLimit = %d in epoch %d", process.ErrInvalidMaxGasLimitPerMetaBlock, gc.maxGasLimitPerMetaBlock, gc.minGasLimit, gasLimitSetting.EnableEpoch)
-	}
-	if gc.maxGasLimitPerMetaMiniBlock < gc.minGasLimit {
-		return nil, fmt.Errorf("%w: maxGasLimitPerMetaMiniBlock = %d minGasLimit = %d in epoch %d", process.ErrInvalidMaxGasLimitPerMetaMiniBlock, gc.maxGasLimitPerMetaMiniBlock, gc.minGasLimit, gasLimitSetting.EnableEpoch)
-	}
-	if gc.maxGasLimitPerTx < gc.minGasLimit {
-		return nil, fmt.Errorf("%w: maxGasLimitPerTx = %d minGasLimit = %d in epoch %d", process.ErrInvalidMaxGasLimitPerTx, gc.maxGasLimitPerTx, gc.minGasLimit, gasLimitSetting.EnableEpoch)
-	}
-
-	return gc, nil
-}
-
-func isPercentageInvalid(percentage float64) bool {
-	isLessThanZero := percentage < 0.0
-	isGreaterThanOne := percentage > 1.0
-	if isLessThanZero || isGreaterThanOne {
-		return true
-	}
-	return false
-}
-
 // SetStatusHandler will set the provided status handler if not nil
 func (ed *economicsData) SetStatusHandler(statusHandler core.AppStatusHandler) error {
```

Comment:
can you add for the status handler a RWMutex?
also in the gasConfigHandler and rewardsConfigHandler

Reply:
added